### ISSUE TITLE
BUG: InvalidCastException on StreamExtensions.WriteObject

### ISSUE THREAD

**Issue by micdenny** [Syroot/BinaryData#9]
this should be hot fixed:

[StreamExtensions.Writing.cs#L787](https://github.com/Syroot/BinaryData/blob/master/src/Syroot.BinaryData/Extensions/StreamExtensions.Writing.cs#L787)
```csharp
else if (type == typeof(UInt64))
{
     Write(stream, (UInt32)value, converter);
}
```

**Comment by RayKoopa** [comment]
Fixed with https://github.com/Syroot/BinaryData/commit/cf831894e3d279f8b1aaeefc27fabd0de8f7102c, released in a moment.

**Comment by micdenny** [comment]
> Fixed with cf83189, released in a moment.

you're flash!! you can close my PR #10
